dijkstra/lib/graph.js

```diff
@@ -7,6 +9,18 @@ function makeEdge(destination, weight) {
   return { destination: destination, weight: weight };
 }
 
+function isValidEdge(node, dest, index) {
+  // avoid self-reference
+  if (dest === index) return false;
+
+  // avoid duplicate edges
+  for (var i=0; i < node.length; i++) {
```

Comment:
Nit: it's better to be consistent about spacing in JS, e.g. `i=0` vs `i = 0`. Usually people go for spaces around operators. The best way to control that (and many other useful things) is by setting up https://eslint.org/. It's easy to just reuse an existing sane config if you're just getting started, e.g. https://github.com/mourner/eslint-config-mourner

Reply:
👍 using your stricter eslint config via https://github.com/freenerd/learning-graph-algorithms/pull/2 and fixing the consistent spacing